File: app/src/main/java/com/naple/android/one_day_one_motivation/activity/SplashActivity.java
package com.naple.android.one_day_one_motivation.activity;

import androidx.appcompat.app.AppCompatActivity;

import android.content.Intent;
import android.content.SharedPreferences;
import android.os.Bundle;
import android.os.Handler;

import com.naple.android.one_day_one_motivation.R;
import com.naple.android.one_day_one_motivation.rest.MongoREST;

import java.util.UUID;

public class SplashActivity extends AppCompatActivity {

    private MongoREST mongoREST = new MongoREST();

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_splash);

        //개인설정 파일 데이터 불러오기
        SharedPreferences sharedPreferences = getSharedPreferences(getString(R.string.shardPreferences), MODE_PRIVATE);
        String uuid = sharedPreferences.getString("uuid", "");
        // 개인설정 파일에 UUID에 대한 데이터가 없다면 생성하여 저장하기
        if (uuid == null || uuid.equals("")) {
            SharedPreferences.Editor editor = sharedPreferences.edit();
            editor.putString("uuid", UUID.randomUUID().toString());
            editor.commit();
        } else {
//            uuid = sharedPreferences.getString("uuid", "");
        }

        Handler handler = new Handler();
        handler.postDelayed(new splashHandler(), 600);
    }

    private class splashHandler implements Runnable {
        public void run() {
            startActivity(new Intent(getApplication(), VideosListActivity.class));
            SplashActivity.this.finish();

        }
    }
}

File: app_kotiln/src/main/java/com/naple/android/one_day_one_motivation/view/splash/SplashActivity.kt
package com.naple.android.one_day_one_motivation.view.splash

import android.content.Intent
import android.os.Bundle
import android.os.Handler
import android.os.Looper
import android.util.Log
import androidx.appcompat.app.AppCompatActivity
import com.naple.android.one_day_one_motivation.R
import com.naple.android.one_day_one_motivation.databinding.ActivitySplashBinding
import com.naple.android.one_day_one_motivation.view.main.MainActivity
import com.naple.android.one_day_one_motivation.view.splash.presenter.SplashContract
import com.naple.android.one_day_one_motivation.view.splash.presenter.SplashPresenter
import java.util.*

class SplashActivity : AppCompatActivity() , SplashContract.View{

    private var presenter : SplashPresenter? = null
    private lateinit var binding: ActivitySplashBinding


    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)

        binding = ActivitySplashBinding.inflate(layoutInflater)

        setContentView(binding.root)

        Handler(Looper.getMainLooper()).postDelayed(
                {startActivity(Intent(application, MainActivity::class.java))}
                ,600)

    }

    override fun onStop() {
        super.onStop()

        //개인설정 파일 데이터 불러오기
        val sharedPreferences = getSharedPreferences(getString(R.string.shardPreferences), MODE_PRIVATE)
        var uuid = sharedPreferences.getString("uuid", "")

        // 개인설정 파일에 UUID 없다면 생성하여 저장하기
        if (uuid == null || uuid == "") {
            val editor = sharedPreferences.edit()
            uuid = UUID.randomUUID().toString()
            editor.putString("uuid", uuid)
            editor.apply()
        } else {
//            uuid = sharedPreferences.getString("uuid", "");
        }
        presenter = SplashPresenter()
        presenter?.loginInsertOrUpdate(uuid)

    }

    override fun onDestroy() {
        super.onDestroy()

        presenter = null
    }
}